### ISSUE TITLE
Issue with the new actor's palette system and updating palette

### ISSUE THREAD

**Issue by quentintamar** [chrismaltby/gb-studio#1020]
**Is your feature request related to a problem? Please describe.**
i've just thought that if i want to use a color palette change on ennemy hit, to add some juice and reward for the actor action (like in oracle of seasons boss impact and other games)
, and there is more than 1 of the same sprite actor in the scene ( and using the same actor's palette because its how its working now) 
All of my actors will be affected by this change and looking like they all have been touched by the player.

**Describe the solution you'd like**
i don't really know how to do that without having specific color palette for each actors like we used to

**Describe alternatives you've considered**
maybe make an "on hit" specific palette that can be applyed to a specific actors with an event

**Additional context**
no screen shot

**Comment by RichardULZ** [comment]
Use anim states to  duplicate your enemy actor animations/directions, but have all sprites on a different palette slot, 
then have scene palette 1 normal, scene palette 2 damaged colour. 
In this way you can also flash enemies by alternating palette slots each frame, and also use OBP0-1 for a slight flash on DMG/SGB.


This is how all games on a gameboy have to be programmed.

Keep in mind this was impossible with standard 2.0b5, where each actor was automatically assigned a palette and slot, there was no way to change the sprite palette at runtime either on the actor slot or scene level due to the complexities and simplifications required.
A mod was released for 2.0b5 (Actor Bits Plugin) to change the palette slot as you would do now with animation states, but a lot of guess work was involved due to automatic palette slot setup per scene. The exposing of scene sprite palette slots is new and only possible with the more manual workflow.

More documentation, tutorials, or guides could be created on achieving this effect explained above, but it is already possible with 3.0 via anim states. Please close this if it solves your issue.

**Comment by quentintamar** [comment]
yeah i know we couldnt change palette on previous version, but a mix of the two could have been nice :)

i thought about using same sprites with reversed colored but i thought it would used too much of the scene frame for no reasons when we could only change palette. but ill take a look on that
